tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11144,6 +11144,51 @@ bool Sema::DiagnoseHLSLMethodCall(const CXXMethodDecl *MD, SourceLocation Loc) {
   return false;
 }
 
+void Sema::DiagnoseSVForLaunchType(const FunctionDecl *FD,
+                                   DXIL::NodeLaunchType LaunchTy) {
+  // Validate Compute Shader system value inputs per launch mode
+  for (ParmVarDecl *param : FD->parameters()) {
+    for (const hlsl::UnusualAnnotation *it : param->getUnusualAnnotations()) {
+      if (it->getKind() == hlsl::UnusualAnnotation::UA_SemanticDecl) {
+        const hlsl::SemanticDecl *sd = cast<hlsl::SemanticDecl>(it);
+        // if the node launch type is Thread, then there are no system values
+        // allowed
+        if (LaunchTy == DXIL::NodeLaunchType::Thread) {
+          if (sd->SemanticName.equals("SV_GroupThreadID") ||
+              sd->SemanticName.equals("SV_GroupIndex") ||
+              sd->SemanticName.equals("SV_GroupID") ||
+              sd->SemanticName.equals("SV_DispatchThreadID")) {
```

Comment:
Question: As long as we're string comparing, can we go broader and simpler by just checking the "SV_" prefix? Do we need to check this at all? Are all semantic decls invalid here?

Reply:
According to the spec, all *compute* System values are invalid. I'll see if just checking SV causes failures in other tests, and if not, leave the change in.

Reply:
I've left both recommended changes in.